app/request_handlers/handlers/lobby.js

```diff
@@ -1,30 +1,47 @@
-const { joinLobby, getPlayersInLobby, clearLobby, getLobbies } = require("../../lobby/lobby.js");
+const { joinLobby, getPlayersInLobby, clearLobby, leaveLobby, getLobbies } = require("../../lobby/lobby.js");
 const { sendToAllClients } = require("../../wss.js");
+const { idGenerator } = require("../../utils.js");
 const { GameReadyData, LobbyListData } = require("../../dataObjects.js");
 const { ANONY_COOKIE_DURATION_MS } = require("../../constants.js");
 const { startGame, lookUpRole } = require("../../game/game.js");
 
+const generateUniqueLobbyId = idGenerator();
+
+const userJoined = {}; // quick access if user is currently in lobby
+
 const startGameIfReady = (isGameReady, lobbyId) => {
     if (isGameReady) {
         const [humanPlayer, aiPlayer] = getPlayersInLobby(lobbyId);
         const gameId = startGame(humanPlayer.username, aiPlayer.username);
         clearLobby(lobbyId);
+		refreshLobbyPage();
         sendToAllClients(GameReadyData(lobbyId, gameId));
     }
 }
 
 const lobbyJoin = (req, res) => {
     const body = req.body;
-    const lobbyId = body.lobbyId;
-    const username = body.username;
-    console.log("POST /lobby/join", body);
+	const username = body.username;
+	if (Object.keys(userJoined).includes(String(username))) { // already in a lobby!
```

Comment:
Instead of writing a line of code with a comment, let's move this line to an `isPlayerInLobby(username)` function. 
Our function names can be our comments :)